sql/ddl_internal.sql

```diff
@@ -267,18 +280,27 @@ CREATE OR REPLACE FUNCTION _timescaledb_internal.drop_chunks_older_than(
 )
     RETURNS VOID LANGUAGE PLPGSQL VOLATILE AS
 $BODY$
+DECLARE 
+    chunk_row _timescaledb_catalog.chunk;
 BEGIN
-    PERFORM _timescaledb_internal.drop_chunk(c.id, false)
-    FROM _timescaledb_catalog.chunk c
-    INNER JOIN _timescaledb_catalog.hypertable h ON (h.id = c.hypertable_id)
-    INNER JOIN  _timescaledb_internal.dimension_get_time(h.id) time_dimension ON(true)
-    INNER JOIN _timescaledb_catalog.dimension_slice ds
-        ON (ds.dimension_id =  time_dimension.id)
-    INNER JOIN _timescaledb_catalog.chunk_constraint cc
-        ON (cc.dimension_slice_id = ds.id AND cc.chunk_id = c.id)
-    WHERE ds.range_end <= older_than_time
-    AND (drop_chunks_older_than.schema_name IS NULL OR h.schema_name = drop_chunks_older_than.schema_name)
-    AND (drop_chunks_older_than.table_name IS NULL OR h.table_name = drop_chunks_older_than.table_name);
+    FOR chunk_row IN SELECT *
```

Comment:
Seems there's a lot of overlap between this function and `truncate_hypertable()`. This has also regressed now since removing `drop_chunk()`. Would be good to make some function that isolates this FOR-loop to one implementation/function that makes the DROP chunk statement. You could even have one loop that accepts a NULL time value to mean "all" chunks from beginning of time. That would cover both use-cases.

Reply:
Yep changed `drop_chunk_older_than` to accept NULL times. Made truncate_hypertable call that function.